### ISSUE TITLE
Freeze when loading project

### ISSUE THREAD

**Issue by JennToo** [kropp/intellij-makefile#109]
I've noticed that if I launch CLion with this plugin enabled, CLion gets stuck on "Loading components". If I disable this plugin, the issue disappears.

Digging through the logs (full logs attached), this bit seems suspicious:

```
2019-11-14 12:08:41,381 [   2990]   INFO - ndex.PrebuiltIndexProviderBase - Prebuilt id-index indices are missing for Python 
2019-11-14 12:08:41,426 [   3035]   INFO - il.indexing.FileBasedIndexImpl - Version has changed for index IdIndex. The index will be rebuilt. 
2019-11-14 12:08:41,438 [   3047]   INFO - il.indexing.FileBasedIndexImpl - Version has changed for index filetypes. The index will be rebuilt. 
2019-11-14 12:08:41,542 [   3151]   INFO - ij.psi.stubs.StubUpdatingIndex - requesting complete stub index rebuild due to changes: added file types:Makefile; 
2019-11-14 12:08:41,543 [   3152]   INFO - il.indexing.FileBasedIndexImpl -  
java.io.IOException
	at com.intellij.psi.stubs.StubVersionMap.updateState(StubVersionMap.java:114)
	at com.intellij.psi.stubs.StubVersionMap.<init>(StubVersionMap.java:56)
	at com.intellij.psi.stubs.StubUpdatingIndex$MyIndex.<init>(StubUpdatingIndex.java:282)
	at com.intellij.psi.stubs.StubUpdatingIndex.createIndexImplementation(StubUpdatingIndex.java:277)
	at com.intellij.util.indexing.FileBasedIndexImpl.createIndex(FileBasedIndexImpl.java:465)
	at com.intellij.util.indexing.FileBasedIndexImpl.initIndexStorage(FileBasedIndexImpl.java:398)
	at com.intellij.util.indexing.FileBasedIndexImpl.registerIndexer(FileBasedIndexImpl.java:359)
	at com.intellij.util.indexing.FileBasedIndexImpl.access$5500(FileBasedIndexImpl.java:99)
	at com.intellij.util.indexing.FileBasedIndexImpl$FileIndexDataInitialization.lambda$initAssociatedDataForExtensions$0(FileBasedIndexImpl.java:2394)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.executeNestedInitializationTask(IndexInfrastructure.java:217)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.runParallelNestedInitializationTasks(IndexInfrastructure.java:205)
	at com.intellij.util.indexing.IndexInfrastructure$DataInitialization.call(IndexInfrastructure.java:166)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

[clion-logs-20191114-121313.zip](https://github.com/kropp/intellij-makefile/files/3848032/clion-logs-20191114-121313.zip)

**Comment by JennToo** [comment]
If I start CLion and I'm fast enough to cancel the project load, I can then load projects normally. This only seems to happen when CLion tries to load the last-used project on startup

**Comment by kropp** [comment]
Thanks for the logs, unfortunately I haven't found anything plugin related in thread freeze dumps.
There seems to be a freeze during the index loading, but there is no plugin code in stack trace.
Disabling plugin leads to rebuilding caches, that may solve the problem.
Did enabling plugin reintroduce the issue? Have you tried the latest CLion & plugin?

**Comment by JennToo** [comment]
Yes re-enabling the plugin reintroduces the problem (or at least it did at the time). I haven't tried the plugin in a few months, so I can try it again in a few days

**Comment by JennToo** [comment]
I tried it with the latest of everything on the same project as before, and the issue does seem to have been resolved

**Comment by kropp** [comment]
Great, thank for the update!

**Comment by angelo-v** [comment]
My Intellij freezes on startup as well after installing this plugin. 

Found this error in the log:

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class name.kropp.intellij.makefile.MakefileParserDefinition
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateExtensionWithPicoContainerOnlyIfNeeded(ComponentManagerImpl.kt:660)
	... 54 more
```

**Comment by kropp** [comment]
@angelo-v what IntelliJ and plugin version do you use? How did you install the plugin?

**Comment by angelo-v** [comment]
oh sorry, no, I had your plugin installed already and installed https://plugins.jetbrains.com/plugin/13740-dockdockbuild. Perhaps they do not harmonize.

I am using Intellij 2020.1 Ultimate

The curious thing is: I have your plugin installed for a long period actually and suddenly IntelliJ proposed to install support for makefiles. The highlighting and green arrows were actually gone. So I installed the proposed DockDockBuild.

No I removed DockDockBuild and the freeze went away, but still IntelliJ complains

> Plugins supporting makefile files not found
